src/notebooks_service.py

```diff
@@ -29,14 +29,24 @@
 import escapism
 import gitlab
 import requests
-from flask import Flask, Response, abort, make_response, redirect, request
+from flask import Flask, Response, abort, make_response, redirect, render_template, request
 from flask import jsonify
 from flask import send_from_directory
 from jupyterhub.services.auth import HubOAuth
 
 SERVICE_PREFIX = os.environ.get('JUPYTERHUB_SERVICE_PREFIX', '/')
 """Service prefix is set by JupyterHub service spawner."""
 
+ANNOTATION_PREFIX = 'hub.jupyter.org'
+"""The prefix used for annotations by the KubeSpawner."""
+
+# check if we are running on k8s
+try:
+    from kubernetes import client, config
```

Comment:
Not sure this test is enough. Running:
```python
config.load_incluster_config()
```
should fail outside of k8s.
You can also see if paths like `/var/run/secrets/kubernetes.io/serviceaccount` exists and are not empty.